### PR TITLE
docs: Make supported language explicit

### PR DESCRIPTION
**Pull request by simonswine**
This makes it explicit what languages are supposed to work and what steps are required to debug problems.

### CODE REVIEW

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md [hunk 1]
```diff
@@ -17,6 +17,12 @@ To use the  `pyroscope.ebpf` component you must run {{< param "PRODUCT_NAME" >}}
 
 You can specify multiple `pyroscope.ebpf` components by giving them different labels, however it's not recommended as it can lead to additional memory and CPU usage.
 
+## Supported languages
+
+This eBPF profiler only collects CPU profiles. Generally natively compiled languages like C/C++, Go, Rust are supported (also see [Troubleshooting]({{< relref "#troubleshooting" >}}) for additional requirements).
```
Comment (clayton-cornell) [inline docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:22]:
```suggestion
This eBPF profiler only collects CPU profiles. Generally, natively compiled languages like C/C++, Go, and Rust are supported. Refer to [Troubleshooting unknown symbols](#troubleshooting-unknown-symbols) for additional requirements.
```

relrefs are deprecated. We should use inline links or define the links at the bottom of the section or page.

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md [hunk 2]
```diff
@@ -17,6 +17,12 @@ To use the  `pyroscope.ebpf` component you must run {{< param "PRODUCT_NAME" >}}
...
+
+The only high-level language supported is Python (as long as `python_enabled=true`). Other high-level languages like Java, Ruby, PHP, JavaScript, etc. will require some additional work in order to correctly show stack traces of methods in these languages. Currently their CPU usage will be displayed belonging to the runtime's methods instead.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:24]:
```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be displayed belonging to the runtime's methods instead.
```

Where is the CPU usage displayed? Who is displaying it? Alloy? Grafana?

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md [hunk 3]
```diff
@@ -113,7 +119,7 @@ If it's not specified, it's attempted to be inferred from multiple sources:
 
 If `service_name` isn't specified and couldn't be inferred, it's set to `unspecified`.
 
-## Troubleshooting unknown symbols
+## Troubleshooting unknown symbols {#troubleshooting}
```
Comment (clayton-cornell) [inline docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:122]:
```suggestion
## Troubleshooting unknown symbols
```

We don't need this if the link is defined as suggested above.

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md (hunk 1 above)
Comment (simonswine) [reply docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:22]: Do you have an example by hand?

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md (hunk 2 above)
Comment (simonswine) [reply docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:24]: I guess Grafana is displaying it in the flamegraphs, maybe we us reported, as that is what Alloy is doing?

```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be reported belonging to the runtime's methods instead.
```

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:22]: Inline linking is like I suggested here.

Links at the bottom of the section/page is used in most topics - for example in `docs/sources/get-started/configuration-syntax/components.md` (we can use relative links or fully qualified URLs.

Comment (simonswine) [reply docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:22]: I see, sorry I wasn't properly reading the suggestion.

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md [hunk 4]
```diff
@@ -17,6 +17,12 @@ To use the  `pyroscope.ebpf` component you must run {{< param "PRODUCT_NAME" >}}
 
 You can specify multiple `pyroscope.ebpf` components by giving them different labels, however it's not recommended as it can lead to additional memory and CPU usage.
 
+## Supported languages
+
+This eBPF profiler only collects CPU profiles. Generally, natively compiled languages like C/C++, Go, and Rust are supported. Refer to [Troubleshooting unknown symbols][troubleshooting] for additional requirements.
+
+Python is the only supported high-level language, as long as `python_enabled=true`. Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly. Currently, their CPU usage will be reported belonging to the runtime's methods instead.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:24]:
```suggestion
Python is the only supported high-level language, as long as `python_enabled=true`.
Other high-level languages like Java, Ruby, PHP, and JavaScript require additional work to show stack traces of methods in these languages correctly.
Currently, the CPU usage for these languages is reported as belonging to the runtime's methods.
```

File: docs/sources/reference/components/pyroscope/pyroscope.ebpf.md [hunk 5]
```diff
@@ -121,7 +127,7 @@ Symbols are extracted from various sources, including:
 - The `.symtab` and `.dynsym` sections in the debug ELF file.
 - The `.gopclntab` section in Go language ELF files.
 
-The search for debug files follows [gdb algorithm](https://sourceware.org/gdb/onlinedocs/gdb/Separate-Debug-Files.html).
+The search for debug files follows [gdb algorithm].
```
Comment (clayton-cornell) [inline docs/sources/reference/components/pyroscope/pyroscope.ebpf.md:130]:
```suggestion
The search for debug files follows [gdb algorithm][].
```

We can add the extra `[]` just for consistency. It's not required but we tend to use it throughout the rest of the docs.

**Review by clayton-cornell — APPROVED** [review]
Looks OK to me. two small tweaks and we are good to go :-)

**Pull request merged**